src/Microsoft.OData.Core/ODataUtils.cs

```diff
@@ -206,11 +206,11 @@ public static ODataServiceDocument GenerateServiceDocument(this IEdmModel model)
 
             ODataServiceDocument serviceDocument = new ODataServiceDocument();
             serviceDocument.EntitySets = model.EntityContainer.EntitySets()
-                .Select(entitySet => new ODataEntitySetInfo() { Name = entitySet.Name, Title = entitySet.Name, Url = new Uri(entitySet.Name, UriKind.RelativeOrAbsolute) }).ToList();

```

Comment:
What's the standard about the usage of "ToList()" after a Linq?

Do you have such investigation? Would you please share detail in to PR description?

Reply:
ToList() is usually expensive O(n) time and O(n) space. But if someone is iterating over the Ienumarable multiple times, its better to use ToList(), In this case we have identified (checked with aGS) that its iterated only 1 time, so better to remove to list.